python/rapidsmpf/rapidsmpf/buffer/resource.pyx

```diff
@@ -183,6 +202,27 @@ cdef class BufferResource:
         if periodic_spill_check is not None:
             period = cpp_Duration(periodic_spill_check)
 
+        # Handle stream pool parameter
+        # If None, create a default pool with 16 streams
+        if stream_pool is None:
+            stream_pool = CudaStreamPool(pool_size=16)
```

Comment:
I'd like to avoid this block entirely, but I wasn't sure how to essentially pass `None` down to the C++ constructor and let us use its default behavior. Maybe that's unavoidable.

But if we do need this, we need to make sure we match the behavior of the `BufferResource` constructor in C++. In particular, pass the non-blocking flag.